Source/WebKit/GPUProcess/graphics/RemoteRenderingBackend.cpp

```diff
@@ -438,6 +466,14 @@ void RemoteRenderingBackend::cacheGradient(Ref<Gradient>&& gradient)
         LOG_WITH_STREAM(DisplayLists, stream << "Received a Gradient without a valid resource identifier");
 }
 
+void RemoteRenderingBackend::releaseGradient(RenderingResourceIdentifier identifier)
+{
+    assertIsCurrent(workQueue());
+    bool success = m_remoteResourceCache.releaseGradient(identifier);
+    MESSAGE_CHECK(success, "Gradientreleased before being cached.");
+}
+
+
 void RemoteRenderingBackend::cacheFilter(Ref<Filter>&& filter)
 {
     ASSERT(!RunLoop::isMain());
```

Comment:
Not really part of this change, but it'd be nice for these checks to be switched to `assertIsCurrent(workQueue())` at some point.

Reply:
Yeah, I have pending patches to do something with them..